modules/templates/htmlrenderer.go

```diff
@@ -43,53 +82,24 @@ func HTMLRenderer(ctx context.Context) (context.Context, *render.Render) {
 	}
 	log.Log(1, log.DEBUG, "Creating "+rendererType+" HTML Renderer")
 
-	compilingTemplates := true
-	defer func() {
-		if !compilingTemplates {
-			return
-		}
-
-		panicked := recover()
-		if panicked == nil {
-			return
-		}
-
-		// OK try to handle the panic...
-		err, ok := panicked.(error)
-		if ok {
-			handlePanicError(err)
-		}
-		log.Fatal("PANIC: Unable to compile templates!\n%v\n\nStacktrace:\n%s", panicked, log.Stack(2))
-	}()
-
-	renderer := render.New(render.Options{
-		Extensions:                []string{".tmpl"},
-		Directory:                 "templates",
-		Funcs:                     NewFuncMap(),
-		Asset:                     GetAsset,
-		AssetNames:                GetTemplateAssetNames,
-		UseMutexLock:              !setting.IsProd,
-		IsDevelopment:             false,
-		DisableHTTPErrorRendering: true,
-	})
-	compilingTemplates = false
+	renderer := &HTMLRender{}
+	if err := renderer.CompileTemplates(); err != nil {
```

Comment:
So it will always compile the templates even if it's in production mode.

Reply:
The `CompileTemplates` is only executed for 2 cases:

1. First time init
2. Dev mode, file watcher reload.